src/Language/PureScript/Linter.hs

```diff
@@ -46,6 +46,7 @@ lint (Module _ _ mn ds _) = censor (addHint (ErrorInModule mn)) $ mapM_ lintDecl
     f (PositionedDeclaration pos _ dec) = addHint (PositionedError pos) (f dec)
     f dec@(ValueDeclaration name _ _ _) = addHint (ErrorInValueDeclaration name) (warningsInDecl moduleNames dec <> checkTypeVarsInDecl dec)
     f (TypeDeclaration name ty) = addHint (ErrorInTypeDeclaration name) (checkTypeVars ty)
+    f (TypeClassDeclaration name args _ _ decs) = addHint (ErrorInTypeClassDeclaration name) (foldMap (checkTypeVarsInDecl' (S.fromList $ fst <$> args)) decs)
```

Comment:
Does it make sense to call `f` recursively on the elements of `decs` here? Then you'd get a `ErrorInTypeDeclaration` wrapper applied too.